metaflow/plugins/kfp/kfp.py

```diff
@@ -580,6 +597,42 @@ def _set_container_resources(
                 mode=mode,
             )
             container_op.add_pvolumes({volume_dir: volume})
+        if kfp_component.accelerator_decorator:
+            accelerator_type: str = kfp_component.accelerator_decorator.attributes[
+                "type"
+            ]
+            if not accelerator_type:
+                raise Exception("You must specify the type of accelerator.")
```

Comment:
thanks for adding the validation, I believe it should be in step_init to be consistent and localized to the decorator, just as is done here:

https://github.com/zillow/metaflow/blob/d115c6fef1816ded198e6c4f57bae83943e1fc70/metaflow/plugins/kfp/kfp_decorator.py#L67-L70